pkg/status/syncratio.go

```diff
@@ -0,0 +1,120 @@
+package status
+
+import (
+	"errors"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// SyncRatio provides an interface for managing syncs which are used to report
+// operator status to CVO.
+type SyncRatio interface {
+	GetSyncs() (failedSyncs int, syncEvents int)
+	ReportFailedSync()
+	ReportSyncEvent()
+	IsSucceeding() (bool, error)
+}
+
+// NewSyncRatio returns a syncRatio object or an error if invalid parameters
+// are provided.
+func NewSyncRatio(successRatio float32, syncsBeforeTruncate int, syncTruncateValue int) (SyncRatio, error) {
+	errMsg := ""
+	if successRatio < 0 || successRatio > 1 {
+		errMsg += "successRatio must be greater than or equal to 0 and less than or equal to 1. "
+	}
+	if syncsBeforeTruncate <= 0 {
+		errMsg += "syncsBeforeTruncate must be greater than 0. "
+	}
+	if syncTruncateValue <= 0 {
+		errMsg += "syncTruncateValue must be greater than 0."
```

Comment:
What is the point in all these error message when you don't do anything with them in the call site?

Reply:
+1

Reply:
That's a good point. Since we are construct the syncRatio with constants in the status package, an error would only happen when a developer changes the constants. I could update the call site to log a fatal error if an error is returned.